internal/plugin/connectors/tcp/mssql/connector_test.go

```diff
@@ -49,6 +50,33 @@ func TestHappyPath(t *testing.T) {
 
 	assert.Equal(t, expectedBackendConn, actualBackendConn)
 }
+
+func TestFailingReadPrelogin(t *testing.T) {
+	logger := logmock.NewLogger()
+	clientConn := mock.NewNetConn(nil)
+	creds := connector.CredentialValuesByID{}
+
+	testError := "injected readPrelogin error"
+	connector := NewSingleUseConnectorWithOptions(
+		logger,
+		nil,
```

Comment:
Don't we want to test that ReadPrelogin fails when all other dependencies succeed? Shouldn't we still mock those our, or will nil be sufficient?

Reply:
I suppose it depends on how black-box-ish we want to be with UT. When the readPreloginRequest fails, it bails out of our Connect() fairly early. So technically, the other dependencies are a don't care for this test. My preference would be to leave these as nil to leave the testing as minimalist as possible.

Reply:
@diverdane this makes the tests less black-box-ish though, which makes them brittle in the event of arbitrary rearrangement of code inside Connect.

Reply:
@BradleyBoutcher, @doodlesbykumbi  I refactored the happy path test and the failing read login tests so that there are default functions being used/shared. So testing is more black box-ish, although I'm sure we can replace what I have with what @doodlesbykumbi is working on.